File: LeetCode/C/590-Unknown-n叉树的后序遍历.cpp
/*
 * @lc app=leetcode.cn id=590 lang=cpp
 *
 * [590] N叉树的后序遍历
 */
#include "main.h"

// @lc code=start
class Solution {
  public:
    vector<int> postorder(Node *root) {
        post(root);
        return res;
    }

  private:
    vector<int> res;
    void post(Node *root) {
        if (root == nullptr)
            return;
        for (auto child : root->children)
            post(child);
        res.push_back(root->val);
    }
};
// @lc code=end

File: LeetCode/Go/590-Unknown-n叉树的后序遍历.go
/*
 * @lc app=leetcode.cn id=590 lang=golang
 *
 * [590] N叉树的后序遍历
 */
package leetcode
// @lc code=start
/**
 * Definition for a Node.
 * type Node struct {
 *     Val int
 *     Children []*Node
 * }
 */

func postorder(root *Node) []int {
	var res = []int{}
	
	var post func(*Node)
	post = func(root *Node){
		if root == nil {
			return
		}
		for _, child := range root.Children {
			post(child)
		}
		res = append(res, root.Val)
	}

	post(root)
	return res
}
// @lc code=end

